### ISSUE TITLE
Pull from base rather than rebase

### ISSUE THREAD

**Issue by Pr0methean** [dependabot/feedback#326]
When Dependabot rebases a PR, Travis-CI and AppVeyor don't always seem to detect that it needs to be checked again. For example, https://github.com/Pr0methean/BetterRandom/pull/100 was merged without testing even though both were enabled and had run on the first push.

A workaround for this might be to provide the option to pull changes from the base branch to an open PR, rather than rebasing.

**Comment by greysteil** [comment]
Ack, sorry about that one (and for the slow reply here).

TL;DR - this shouldn't happen again. Let me know if it does.

Long version: GitHub's checks API makes it difficult to know whether a `queued` check will ever return. If the app that GitHub has requested run the check never responds (because it's not configured on that app) then the check remains as `queued` in the API indefinitely. To stop that causing problems, Dependabot used to ignore all `queued` checks.

I've just improved the logic above by making Dependabot keep a record of all check types that have ever returned on a repo, and using that to determine whether a check is likely to return. If it is, and it's `queued`, it will no longer be ignored and will block automerging (as it should).

Apologies again, and thanks for reporting this!

**Comment by Pr0methean** [comment]
When you don't have data on a particular check one way or the other, might
it be better to use a timeout of, say, 48 hours?
